### PR TITLE
repaired xtend build. workaround for eclipse/xtext#1976

### PR DESCRIPTION
**Pull request by cdietrich**
repaired xtend build. workaround for eclipse/xtext#1976
Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -39,8 +39,24 @@ subprojects {
 	apply from: "${rootDir}/gradle/xtend-compiler-settings.gradle"
 	apply from: "${rootDir}/gradle/maven-deployment.gradle"
 	apply from: "${rootDir}/gradle/manifest-gen.gradle"
+
+	configurations.all {
+		if (name.endsWith("ToolingMain") || name.endsWith("ToolingTest")) {
```
Comment (oehme) [inline build.gradle:44]: Why not `contains("Tooling")`?

Comment (cdietrich) [reply build.gradle:44]: should also work.
unfortunately this workaround that works local, does not work on server

Comment (cdietrich) [reply build.gradle]: now it seems to work fine

File: build.gradle [hunk 2]
```diff
@@ -39,6 +39,21 @@ subprojects {
 	apply from: "${rootDir}/gradle/xtend-compiler-settings.gradle"
 	apply from: "${rootDir}/gradle/maven-deployment.gradle"
 	apply from: "${rootDir}/gradle/manifest-gen.gradle"
+
+	configurations.all {
+		if (name.contains("Tooling")) {
+			resolutionStrategy { 
+				eachDependency { DependencyResolveDetails details ->
+					if (details.requested.group == 'org.eclipse.platform' && details.requested.name == 'org.eclipse.core.runtime') {
+						details.useVersion "3.19.0"
```
Comment (jonahgraham) [inline build.gradle:48]: Is the syntax difference here to line 51 `(` and `)` important.

**Review by jonahgraham — CHANGES_REQUESTED** [review]
@cdietrich I am pleased that this makes the build work - but from the commit comment I don't understand why it was broken, nor why this can't be solved by adding versions to https://github.com/eclipse/lsp4j/blob/85273590aea17a136e37a633a777682c3f36c3e4/gradle/versions.gradle#L15-L18 ? I have read through the extremely long discussions you have had to try to figure this out in https://github.com/eclipse/xtext/issues/1976

Can you please add a summary/comment somewhere so that this can be maintained going forward.

IIUC something like:

> We need to keep some bundles on the last version that works with Java 8 so we can continue to build and target Java 8 with LSP4J.

Additionally, are these dependencies build time only, or under what circumstance does it become a runtime dep too? Does this relate to the removal of runtime deps on xbase in #528 and #529?

Thanks.

**Comment by cdietrich** [comment]
summary. platform and jdt have version ranges. these version ranges are considered by gradle to bee good and that its a good idea to pull the latest greatest core.runtime and equinix.common versions. these run with java 11 only but the build here runs with java 8, so this will explode. 

so here i fix it by telling gradle to pull the versions are are intended to be used by gradle.

an alternative solution would be to run the build with java 11 against java 8 target

File: build.gradle (hunk 2 above)
Comment (cdietrich) [reply build.gradle:48]: nope they are not, will harmonize

**Review by jonahgraham — APPROVED** [review]
Thanks @cdietrich for the summary.

File: build.gradle [hunk 3]
```diff
@@ -39,6 +39,21 @@ subprojects {
 	apply from: "${rootDir}/gradle/xtend-compiler-settings.gradle"
 	apply from: "${rootDir}/gradle/maven-deployment.gradle"
 	apply from: "${rootDir}/gradle/manifest-gen.gradle"
+
+	configurations.all {
+		if (name.contains("Tooling")) {
+			resolutionStrategy { 
+				eachDependency { DependencyResolveDetails details ->
+					if (details.requested.group == 'org.eclipse.platform' && details.requested.name == 'org.eclipse.core.runtime') {
```
Comment (oehme) [inline build.gradle:47]: Can you add the xtext-dev-bom here instead of a resolution strategy?

Comment (cdietrich) [reply build.gradle:47]: Will give it a try tomorrow

Comment (cdietrich) [reply build.gradle:47]: something like https://github.com/eclipse/lsp4j/compare/cd_repairBuild_variantB @oehme ?

Comment (oehme) [reply build.gradle:47]: Yep :+1:

Comment (cdietrich) [reply build.gradle]: updated the PR

**Review by jonahgraham — APPROVED** [review]
I simply don't know gradle well enough - but this looks nice and clean and I like how in this revision the hard coded version numbers are gone.

**Comment by cdietrich** [comment]
@jonahgraham  you can find them here https://github.com/eclipse/xtext-lib/blob/d8f2aba7be9b515cf53e31d820d2f8bcd00de97d/org.eclipse.xtext.dev-bom/build.gradle#L64

**Comment by jonahgraham** [comment]
> @jonahgraham you can find them here https://github.com/eclipse/xtext-lib/blob/d8f2aba7be9b515cf53e31d820d2f8bcd00de97d/org.eclipse.xtext.dev-bom/build.gradle#L64

That is a good place for them :-) thanks for the link

**Pull request merged**